### PR TITLE
Set up example API endpoint and Database class.

### PR DESCRIPTION
**Pull request by howard-yen**
# Goal 

* In this PR, I updated `app.py` and created `database.py` with examples of how to set up API endpoints and associated logging steps. 
* As a result, you should be able to follow the example I created and make any endpoint (in `app.py`) and its associated database queries/methods (in `database.py`).

# Testing

* The example endpoint `/api/test-database` can be tested by first starting the server with `flask run`, and then manually going to that endpoint. You will see the expected data there. 

# More Details

* In `app.py`, you will see that there is an API routes section that will contain all the api endpoints. Currently, there is a `/api/test-database` endpoint that calls the function `test_database()`
* `database.py` creates a Database object that will handle all the querying to the mongo database. Currently, there is a test function `get_all_test()` that gets all the people in the test database.
* You will also need to update the `.env` file in the base directory with the `MONGO` access string that is also in the `back_end/.env`. 

# TODO
* merge with #11 's database.py

### CODE REVIEW

**Review by jkuo168 — COMMENTED** [review]
Approved!

**Review by shannon-heh — APPROVED** [review]
lgtm, only some small suggestions!

File: app.py [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+# ------------- API ROUTES ------------- #
+"""
+    When writing the API endpoints, refer to the test-database
+    endpoint below as an example.
+"""
+
+@app.route("/api/test-database", methods=["GET"])
+def test_database():
+    logger.info("Reached /api/test-database")
+    return jsonify(db.get_all_test())
```
Comment (shannon-heh) [inline app.py:47]: going to this endpoint is returning `TypeError: Object of type Cursor is not JSON serializable`. 

need to convert cursor to list before jsonifying. also as you mentioned in `database.py`, make sure to remove `ObjectId` field before jsonifying too.

**Review by shannon-heh — CHANGES_REQUESTED** [review]
sorry, taking back my previous approval. primarily to request fix to `test_database` endpoint so it doesn't error for others.

File: database.py [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+    def __init__(self):
+        self.connect()
+        self.db_test = self.client.test
+
+    def connect(self):
+        self.client = MongoClient(os.getenv("MONGO"), tlsCAFile=certifi.where())
+        self.db = self.client.admin
+        logger.info(f"MongoDB server status: {self.db.command('serverStatus')}")
+
+    def as_list(self, cursor):
```
[the remark this answers is not in the document]
Comment (howard-yen) [reply database.py:30]: oh I didn't know that, thanks so much!

File: back_end/scrape_eval.py [hunk 3]
```diff
@@ -1,11 +1,44 @@
 import requests
 import browser_cookie3
 from bs4 import BeautifulSoup
+from dotenv import load_dotenv
+import os
+
+load_dotenv()
 
 BASE_URL = "https://registrarapps.princeton.edu/course-evaluation?ssub={}&courseinfo={}&terminfo={}"
 USER_AGENT = "Mozilla/5.0 (X11; Linux x86_64; rv:93.0) Gecko/20100101 Firefox/93.0"
```
[the remark this answers is not in the document]
Comment (howard-yen) [reply back_end/scrape_eval.py:10]: right, so this is from a while back and technically shouldn't be part of this pr... but i'll change it to resinde-course-selection

File: app.py (hunk 1 above)
Comment (howard-yen) [reply app.py:47]: fixed

**Review by shannon-heh — APPROVED** [review]
lgtm!

**Pull request merged**